infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/AbstractWorkspaceServiceAccount.java

```diff
@@ -127,19 +132,31 @@ private void ensureImplicitRolesWithBindings(Client k8sClient) {
         k8sClient,
         METRICS_ROLE_NAME,
         Arrays.asList("pods", "nodes"),
+        null,
         singletonList("metrics.k8s.io"),
         Arrays.asList("list", "get", "watch"),
         serviceAccountName + "-metrics");
+
+    // credentials-secret role
+    ensureRoleWithBinding(
+        k8sClient,
+        SECRETS_ROLE_NAME,
+        singletonList("secrets"),
+        singletonList(CREDENTIALS_SECRET_NAME),
+        singletonList(""),
+        Arrays.asList("get", "patch"),
+        serviceAccountName + "-secrets");
   }
 
   private void ensureRoleWithBinding(
       Client k8sClient,
       String roleName,
       List<String> resources,
+      @Nullable List<String> resourceNames,
```

Comment:
can we pass empty list instead of having this nullable?

Reply:
I think it is better to keep this parameter as `Nullabale` because in compare with other parameters witch are mandatory,  this is an optional parameter.  @skabashnyuk @sleshchenko WDYT?

Reply:
I would consider here, instead of `resources, resourceNames, ...` arguments, replace it with the Role (it means `R` here), and let the caller build the role with `buildRole`. I think it would make the "API" cleaner, although it's used only privately.

This would only move this "Nullable" discussion one level lower though, but it would reduce number of "Nullable" arguments in the code. For `buildRole`, in theory the correct way how to do it would be to implement own generic `RoleBuilder` that could do both k8s and openshift Roles. But I think it is overkill. I still think that emptyList() is better than null.

Reply:
Replaced with the role object instead of passing all parameters, but I still think that `Nullable` is more suitable for the role creation method.

Reply:
thanks